/* Copyright 2020 Istio Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package istio.wasm.example.opa;

// Input for opa policy engine
message OpaPayload {
  message Input {
    // Principal of source workload.
    string source_principal = 1;

    // FQDN of destination service.
    string destination_service = 2;

    // Request Operation.
    string request_operation = 3;

    // URL path of the request.
    string request_url_path = 4;
  }

  // OPA input
  Input input = 1;
}

// Response of opa policy engine
message OpaResponse { bool result = 1; }